web/thesauruses/javascript/ECMAScript 2020/classes.json

```diff
@@ -0,0 +1,116 @@
+{
+  "meta": {
+    "language": "javascript",
+    "language_version": "ECMAScript 2020",
+    "language_name": "JavaScript",
+    "structure": "classes"
+  },
+  "concepts": {
+    "normal_class": {
+      "code": "class ClassName {\n // Class body containing variables and methods \n}",
+      "comment": "Newer versions of Javascript permit class but are prototypes (an alternative approach to the object concept) which is the core concept of Javascript.",
+      "name": "Normal class"
+    },
+    "abstract_class": {
+      "not-implemented": true,
+      "name": "Abstract class"
+    },
+    "interface": {
+      "not-implemented": true,
+      "name": "Interface"
+    },
+    "read_only_class": {
+      "not-implemented": true,
+      "name": "Read-only class"
+    },
+    "static_class": {
+      "not-implemented": true,
+      "name": "Static class"
+    },
+    "inner_class": {
+      "not-implemented": true,
+      "name": "Inner class"
+    },
+    "packages": {
+      "comment": "A package manager like NPM (https://www.npmjs.com/) or Deno (https://deno.land/) can be used to create and manage Javascript packages.",
+      "not-implemented": true,
+      "name": "Packages"
+    },
+    "class_with_generic_type": {
+      "not-implemented": true,
+      "name": "Class with a generic type"
+    },
+    "private_variables": {
+      "code": "class ClassName {\n #privateField = 0;\n}",
```

Comment:
You don't need a var/let in front of #privatefield?

Reply:
It's a new feature from ECMAScript 2023 (I made a mistake by putting it in the 2020 and 2021 versions, I will correct that)

References :
- Private class fields on MDN : https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes/Private_class_fields
- Finished proposals of ECMAScript 2023 : https://github.com/tc39/proposals/blob/HEAD/finished-proposals.md#finished-proposals